Healthcare-Hospital-Management-System/Healthcare-Hospital-Management-System/Services/DoctorService.cs

```diff
@@ -1,66 +1,52 @@
-using System.Collections.Generic;
 using HealthcareHospitalManagementSystem.Models;
+using HealthcareHospitalManagementSystem.Services;
+using HealthcareHospitalManagementSystem.Infrastructure;
+using System;
+using System.Collections.Generic;
 
-namespace HealthcareHospitalManagementSystem.Services
+public class DoctorService : IDoctorService
 {
-    public class DoctorService : IDisposable, IDoctorService
+    private List<Doctor> _doctors;
+    private const int MaxDoctors = 50;
+    public static int TotalDoctorsAdded { get; private set; } = 0;
+    public Logger Logger { get; set; }
+
+    public DoctorService()
     {
-        private readonly List<Doctor> _doctors = new List<Doctor>();
-        private readonly FileStream _transactionLogFileStream;
-        private bool _disposed = false;
+        _doctors = new List<Doctor>();
+    }
 
-        public DoctorService()
-        {
-            
-        }
-        
-        public DoctorService(string filePath)
-        {
-            _transactionLogFileStream = new FileStream(filePath, FileMode.OpenOrCreate);
-        }
+    public void AddDoctor(Doctor doctor, Logger logger)
+    {
+        string message;
 
-        public void LogTransaction(string message)
+        if (_doctors.Count >= MaxDoctors)
         {
-            if (_disposed)
-                throw new ObjectDisposedException("DoctorService");
-            
-            byte[] messageBytes = System.Text.Encoding.UTF8.GetBytes($"DateTime: {DateTime.UtcNow}, Message: {message}\n");
-            _transactionLogFileStream.Write(messageBytes, 0, messageBytes.Length);
+            message = $"Cannot add doctor {doctor.Name}. Maximum number of doctors reached.";
+            Logger?.LogError(message);
```

Comment:
You are using Logger defined at line 12, as property. 
But you do not use logger passed as parameter in the method. I see in the previous commit you used a parameter but then for some reason you changed it to the property.

By assignment requirements, you need to have both at the same time.

For now, you have unused parameter in the method - Logger logger